Libraries/pbxbuild/Sources/Tool/ClangResolver.cpp

```diff
@@ -242,6 +249,7 @@ resolveSource(
     Tool::CompilerCommon::AppendIncludePathFlags(&arguments, env, toolContext->searchPaths(), headermapInfo);
     AppendFrameworkPathFlags(&arguments, env, toolContext->searchPaths());
     AppendCustomFlags(&arguments, env, fileType->GCCDialectName());
+    AppendPreprocessorDefinitionFlags(&arguments, env);
```

Comment:
Could you change `SwiftResolver` to use this function as well rather than duplicating the logic?